src/accelerate/accelerator.py

```diff
@@ -230,6 +245,51 @@ def prepare_model(self, model):
             model.forward = torch.cuda.amp.autocast()(model.forward)
         return model
 
+    def _prepare_deepspeed(self, *args):
+
+        ds_plugin = self.state.deepspeed_plugin
+        self.ds_config = ds_plugin.ds_config
+
+        batch_size = [obj.batch_size for obj in args if hasattr(obj, "batch_size")]
+        assert len(batch_size) > 0, "You must specify training_dataloader in `accelerate.prepare()` when using DeepSpeed"
+        logger.info("Since you passed both train & eval dataloader, `is_train_batch_min` will decide the `train_batch_size`")
+        batch_size = min(batch_size) if ds_plugin.is_train_batch_min else max(batch_size)
+
+        self.ds_config["train_batch_size"] = batch_size * ds_plugin.gradient_accumulation_steps * self.num_processes
+
+        result = [self._prepare_one(obj) if isinstance(obj, torch.utils.data.DataLoader) else obj for obj in args]
+
+        model = None
+        optimizer = None
+        model_parameters = None
+        for obj in result:
+            if isinstance(obj, torch.nn.Module):
+                model = obj
+                model_parameters = filter(lambda p: p.requires_grad, model.parameters())
+            elif isinstance(obj, (torch.optim.Optimizer, dict)):
+                optimizer = obj
+
+        # TODO: how dict based optimizer should be passed (in optimizer or ds_plugin) ??
+        # if isinstance(optimizer, dict):
+        #     self.ds_config.update({"optimizer": optimizer})
+        #     optimizer = None
+
+        # useful when only eval_dataloader is given into `accelerator.prepare()`
+        if model is not None:
+            model, optimizer, _, _ = deepspeed.initialize(model=model, optimizer=optimizer, config_params=self.ds_config, dist_init_required=False, model_parameters=model_parameters)
+            assert isinstance(model, deepspeed.DeepSpeedEngine), "PipelineEngine not supported currently"
+            for i in range(len(result)):
+                if isinstance(result[i], torch.nn.Module):
+                    result[i] = DeepSpeedEngineWrapper(model)
+                elif isinstance(result[i], torch.optim.Optimizer):
+                    result[i] = DeepSpeedOptimizerWrapper(optimizer, model)
+            self.model = model # pointing for model.backward()
+        
+        if self.distributed_type == DistributedType.DEEPSPEED:
+            assert hasattr(self, "model"), "Accelerator instance must have model as its attribute"
```

Comment:
The error message should be more along the lines of "You need to pass the model along the optimizer when using Deepspeed."